src/rng.rs

```diff
@@ -86,12 +177,12 @@ impl<D: Digest> FiatShamirRng for FiatShamirChaChaRng<D> {
 
     /// Create a new `Self` by initializing with a fresh seed.
     #[inline]
-    fn from_seed<'a, T: 'a + ToBytes>(seed: &'a T) -> Self {
-        let mut bytes = Vec::new();
-        seed.write(&mut bytes).unwrap_or(());
-        let seed = D::digest(&bytes);
+    fn from_seed(seed: <Self::Seed as FiatShamirRngSeed>::FinalizedSeed) -> Self {
```

Comment:
About the derivation of ``r_seed`` from ``seed``, line 183:
What if the digest has an output length which is incompatible to the demanded 32 bytes key/seed of ChaCha? Does the ``.unwrap_or([0u8; 32])`` still output the zero seed then? 
If yes, we should prevent the usage of incompatible digests.

Reply:
Done, put a "static assert"